lib/postscript/request.ex

```diff
@@ -70,11 +62,14 @@ defmodule Postscript.Request do
   end
 
   defp do_retry(response, request, config) do
-    attempt = Map.get(request.private, :attempt)
+    attempt = Map.get(request.private, :attempt, 0)
 
     max_attempts = Keyword.get(config.retry_opts, :max_attempts, 3)
 
     if max_attempts > attempt do
+      private = Map.put(request.private, :attempt, attempt + 1)
+      request = Map.put(request, :private, private)
```

Comment:
can we use `put_in`?

```elixir
request = put_in(request, [:private, :attempt], attempt + 1)
```

Reply:
I tried that, but `put_in` doesn't like the request struct. I think it's only for maps?